File: 0135-candy/0135-candy.cpp
class Solution {
public:
    int candy(vector<int>& ratings) {
        int n = ratings.size();
        vector<int> CandyArr(n,1);
        int TotalCandy = 0;
        
        //scan from left to right
        for(int i=0; i<n-1; i++){
            if(ratings[i] < ratings[i+1]){
                CandyArr[i+1] = CandyArr[i] + 1; 
            }
        }
        //Scan from right to left
        for(int i=n-1; i>0; i--){
            if( ratings[i-1] > ratings[i]){
                if(CandyArr[i-1] <= CandyArr[i])
                    CandyArr[i-1] = CandyArr[i] + 1;
            }
        }
            
        for(int i=0; i<n; i++){
            TotalCandy += CandyArr[i];
        }
        
        return TotalCandy;
    }
};

File: 0135-candy/0135-candy.java
class Solution {
    public int candy(int[] ratings) {
        int n = ratings.length;
        int[] candyArr = new int[n];
        Arrays.fill(candyArr,1);

        for(int i=0; i<n-1; i++){
            if(ratings[i+1] > ratings[i]){
                candyArr[i+1] = candyArr[i]+1;
            }
        }

        for(int i= n-1; i>0; i--){
            if(ratings[i-1] > ratings[i]){
                if(candyArr[i-1] <= candyArr[i]){
                    candyArr[i-1] = candyArr[i]+1;
                }
            }
        }
        
        int ans = 0;
        for(int i=0; i<n; i++){
            ans += candyArr[i];
        }

        return ans;
    }
}